indexer/src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -209,27 +209,37 @@ class BackgroundExternalDatabaseSyncingService(
         }.toSet()
     }
 
-    private fun removeNonexistentTablesAndColumnsForOrg(orgId: UUID, existingTableIds: Set<UUID>, existingColumnIds: Set<UUID>) {
+    private fun removeNonexistentTablesAndColumnsForOrg(
+        orgId: UUID,
+        existingTableIds: Set<UUID>,
+        existingColumnIds: Set<UUID>
+    ) {
 
         // delete missing tables
 
-        val tableIdsToDelete = organizationExternalDatabaseTables.keySet(Predicates.equal(ORGANIZATION_ID_INDEX, orgId)).filter {
-            !existingTableIds.contains(it)
-        }.toSet()
+        val tablesToDelete: Set<OrganizationExternalDatabaseTable> = organizationExternalDatabaseTables
```

Comment:
why are you loading the full table objects here?

Reply:
As it's currently implemented, I need to pass the table objects to `deleteDatasets` below because I need to access the table `oid`. This will probably change depending on what we decide to do with the `oid` issue in the discussion above.

Reply:
Fixed.